payments_service/auth/tests/test_auth.py

```diff
@@ -50,6 +68,34 @@ def test_view_access_denied(self):
             'You do not have permission to perform this action.')
 
     def test_view_access_granted(self):
-        self.prepare_session(buyer_uuid='some-uuid', buyer_pk=1)
+        self.login()
         res, data = self.get()
         eq_(res.status_code, 200, res)
+
+    def test_view_requires_csrf_token(self):
+        self.login()
+        self.client.handler.enforce_csrf_checks = True
+        res, data = self.post({})
+        eq_(res.status_code, 403, res)
+
+    def test_can_access_view_with_csrf_token(self):
+        self.login()
+        self.client.handler.enforce_csrf_checks = True
+
+        # Simulate CSRF middleware processing within Django.
+        fake_request = HttpRequest()
+        fake_response = HttpResponse()
+        middleware = csrf.CsrfViewMiddleware()
+        # Generate a CSRF token.
+        middleware.process_view(fake_request, lambda: None, [], {})
+        # Simulate exposing a CSRF token on a web form. This has a
+        # side effect of preparing the middleware for response processing
+        # since CSRF processing is enabled lazily.
+        csrf_token = csrf.get_token(fake_request)
+        # Set a CSRF cookie.
+        middleware.process_response(fake_request, fake_response)
+        # Prepare to send the cookie in our test client's request.
+        self.client.cookies.update(fake_response.cookies)
```

Comment:
The "server side persistence" is the cookie so we need to send a cookie to the protected view as well as a token header.